### PR TITLE
feat: implement SideroLink

### LINKED ISSUE #615: SideroLink: server implementation
See also talos-systems/talos#4448.

We can spin up user-space Wireguard implementation in the container.

Sidero tunnel address is fixed for all clients. Node tunnel address could be derived from the node UUID. 

Sidero should keep its own Wireguard key in a `Secret`, and also keep track of `Server` public Wireguard keys (keep it in the `Status` of `ServerBinding`).

Sidero configures Wireguard based on this information.

Wireguard endpoint can be exposed either using host networking or via load balancer (like TFTP).

Implement SideroLink API (add to the Sidero API).

### PR DESCRIPTION
**Pull request by smira**
Fixes #615

This implements the SideroLink integration on Sidero side. This feature
requires Talos 0.14 from master as of right now.

SideroLink status is stored in `ServerBinding` resource for each server.

Signed-off-by: Andrey Smirnov <andrey.smirnov@talos-systems.com>

### CODE REVIEW

File: app/sidero-controller-manager/cmd/siderolink-manager/server.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/cluster-api/util/patch"
+	runtimeclient "sigs.k8s.io/controller-runtime/pkg/client"
+
+	pb "github.com/talos-systems/siderolink/api/siderolink"
+
+	sidero "github.com/talos-systems/sidero/app/caps-controller-manager/api/v1alpha3"
+)
+
+// Server implents gRPC API.
```
Comment (Unix4ever) [inline app/sidero-controller-manager/cmd/siderolink-manager/server.go:27]:
```suggestion
// Server implements gRPC API.
```

File: Dockerfile [hunk 2]
```diff
@@ -45,7 +45,7 @@ ENV GOMODCACHE /.cache/mod
 RUN --mount=type=cache,target=/.cache go install sigs.k8s.io/controller-tools/cmd/controller-gen@v0.6.2
 RUN --mount=type=cache,target=/.cache go install k8s.io/code-generator/cmd/conversion-gen@v0.21.3
 RUN --mount=type=cache,target=/.cache go install mvdan.cc/gofumpt/gofumports@v0.1.1
-RUN curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b /toolchain/bin v1.38.0
+RUN curl -sfL https://install.goreleaser.com/github.com/golangci/golangci-lint.sh | bash -s -- -b /toolchain/bin v1.43.0
```
Comment (smira) [inline Dockerfile:48]: had to bump linter, as old one can't lint wireguard code :hankey:

File: app/sidero-controller-manager/config/manager/manager.yaml [hunk 3]
```diff
@@ -90,4 +103,41 @@ spec:
...
+              cpu: 500m
+              memory: 512Mi
+            requests:
+              cpu: 50m
+              memory: 128Mi
+          securityContext:
+            capabilities:
+              add:
+                - NET_ADMIN
+            privileged: false
```
Comment (smira) [inline app/sidero-controller-manager/config/manager/manager.yaml:134]: not doing privileged, just mapping `/dev/net/tun` to make it possible to create `tun` devices for Wireguard

File: sfyra/pkg/tests/server_class.go [hunk 4]
```diff
@@ -39,8 +39,8 @@ import (
 )
 
 const (
-	defaultServerClassName  = "default"
-	workloadServerClassName = "workload"
+	defaultServerClassName = "default"
+	serverClassName        = "sfyra"
```
Comment (smira) [inline sfyra/pkg/tests/server_class.go:43]: using `sfyra` for all deployed clusters, as this allows to test with Talos `HEAD`

**Review by frezbo — APPROVED** [review]

**Review by Unix4ever — APPROVED** [review]

**Pull request merged**